src/persistent-handle-cleanup.cc

```diff
@@ -0,0 +1,36 @@
+#include "persistent-handle-cleanup.h"
+
+#include "v8.h"
+
+#include "env.h"
+#include "env-inl.h"
+#include "smalloc.h"
+#include "node-contextify.h"
+
+namespace node {
+
+using v8::Persistent;
+using v8::Value;
+using v8::Isolate;
+using smalloc::CallbackInfo;
+
+void PersistentHandleCleanup::VisitPersistentHandle(
+    Persistent<Value>* value, uint16_t class_id) {
+  Isolate::DisallowJavascriptExecutionScope scope(env()->isolate(),
+      Isolate::DisallowJavascriptExecutionScope::CRASH_ON_FAILURE);
+
+  if (value->IsWeak()) {
+    switch (static_cast<ClassId>(class_id)) {
+      case SMALLOC:
```

Comment:
Just for future note. Will be able to take this out in the not too distant future.